File: src/1941.check-if-all-characters-have-equal-number-of-occurrences.2053/solution.ts
function areOccurrencesEqual (s: string): boolean {
    const counts:number[] = new Array(26).fill(0);
    for (let i = 0; i < s.length; i++) {
        counts[s.charCodeAt(i) - 97]++;
    }
    let index = 0;
    let frequency = 0;
    while (index < 26) {
        if (counts[index]) {
            frequency = counts[index++];
            break;
        } else {
            index++;
        }
    }
    while (index < 26) {
        if (counts[index] === 0) {
            index++;
            continue;
        }
        if (counts[index] !== frequency) {
            return false;
        }
        index++;
    }
    return true;
}

File: src/1941.check-if-all-characters-have-equal-number-of-occurrences.2053/solution.java
class Solution {
    public boolean areOccurrencesEqual(String s) {
        int[] counts = new int[26];
        for(char c : s.toCharArray()){
            counts[c-'a']++;
        }
        int base = 0;
        for(int c : counts){
            if(c == 0){
                continue;
            }
            if(base == 0){
                base = c;
                continue;
            }
            if(c != base){
                return false;
            }
        }
        
        return true;
    }
}